hawkular-command-gateway/hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/bus/BusEndpointProcessors.java

```diff
@@ -222,8 +221,8 @@ protected void onBasicMessage(BasicMessageWithExtraData<BasicMessage> messageWit
     @Inject
     private BusCommandContextFactory commandContextFactory;
 
-    @Resource(name = Constants.CONNECTION_FACTORY_JNDI)
-    private ConnectionFactory connectionFactory;
+    @Inject
+    private BusConnectionFactoryProvider connectionFactoryProvider;
```

Comment:
> How was it previously done? Was it with @Singleton @Startup EJB?

There was no change in this area of Command Gateway since a couple of months. The only difference that occurred recently is that Accounts have gone away. Maybe Accounts just caused some lucky coincidence (e.g. a mere delay by deploying accounts.war before cmdgw.war) that prevented this from happening, because as I said, usually the message in container log reporting that the JMS objects were bound to the respective names appears immediately after the NamingException thrown from the cmdgw deployment.

Indeed, it looks like CDI does not grant anything about the order of an `@ApplicationScoped` bean's initializer and the availability of the `@Resource` s. I have no idea if `@Singleton @Startup` gives such guarantees. 

> I'm not aware of any limits from CDI. My guess is that it would just wait.

I think that what you propose would make the deployment potentially slower because of the waiting during the injection that happens during the deployment. My implementation eliminates the waiting as far as possible through querying JNDI lazily as late as possible - i.e. when the ConnectionFactory is actually used on a client's connect. I admit the present solution is not elegant but it works well enough and I prefer not to spend more time on this.